common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -418,6 +415,47 @@ static boolean equals(byte[] bytes1, int startPos1, byte[] bytes2, int startPos2
         }
     }
 
+    static int equalsConstantTime(byte[] bytes1, int startPos1, byte[] bytes2, int startPos2, int length) {
+        long result = 0;
+        final long baseOffset1 = BYTE_ARRAY_BASE_OFFSET + startPos1;
+        final long baseOffset2 = BYTE_ARRAY_BASE_OFFSET + startPos2;
+        final int remainingBytes = length & 7;
+        final long end = baseOffset1 + remainingBytes;
+        for (long i = baseOffset1 - 8 + length, j = baseOffset2 - 8 + length; i >= end; i -= 8, j -= 8) {
+            result |= UNSAFE.getLong(bytes1, i) ^ UNSAFE.getLong(bytes2, j);
```

Comment:
@buchgr - Thanks for the details. I agree it doesn't seem practical to do this in java and may be more trouble than it is worth for this use case anyways.